migration-engine/connectors/sql-migration-connector/src/sql_destructive_changes_checker/destructive_check_plan.rs

```diff
@@ -43,14 +45,24 @@ impl DestructiveCheckPlan {
     ) -> SqlResult<DestructiveChangeDiagnostics> {
         let mut results = DatabaseInspectionResults::default();
 
-        for unexecutable in &self.unexecutable_migrations {
-            self.inspect_for_check(unexecutable, &mut results, schema_name, conn)
-                .await?;
-        }
+        let inspection = async {
+            for unexecutable in &self.unexecutable_migrations {
+                self.inspect_for_check(unexecutable, &mut results, schema_name, conn)
+                    .await?;
+            }
 
-        for warning in &self.warnings {
-            self.inspect_for_check(warning, &mut results, schema_name, conn).await?;
-        }
+            for warning in &self.warnings {
+                self.inspect_for_check(warning, &mut results, schema_name, conn).await?;
+            }
+
+            Ok::<(), SqlError>(())
+        };
+
+        // Ignore the timeout error, we will still return useful warnings.
+        match timeout(destructive_check_timeout_duration(), inspection).await {
+            Ok(Ok(())) | Err(Elapsed { .. }) => (),
+            Ok(Err(err)) => return Err(err),
+        };
```

Comment:
Not a fan of this, but I guess as long as we're using Tokio we can't do better. Once we have [`async_std::Future::timeout`](https://docs.rs/async-std/1.6.2/async_std/future/trait.Future.html#method.timeout) available we could probably make this flow a bit better.